File: database/src/main/java/hr/foi/air/database/entities/Participation.java
package hr.foi.air.database.entities;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.TypeConverters;

import java.util.Date;

import hr.foi.air.database.converters.DateConverter;

@Entity(tableName = "participations", primaryKeys = {"player_id", "event_id"})
@TypeConverters(DateConverter.class)
public class Participation {
    @ForeignKey(entity = User.class, parentColumns = "id", childColumns = "player_id")
    @ColumnInfo(index = true, name = "player_id")
    int playerId;

    @ForeignKey(entity = Event.class, parentColumns = "id", childColumns = "event_id")
    @ColumnInfo(index = true, name = "event_id")
    int eventId;

    @ForeignKey(entity = Chat.class, parentColumns = "id", childColumns = "chat_id")
    @ColumnInfo(name = "chat_id")
    int chatId;

    @ColumnInfo(name = "join_date")
    Date joinDate;

    public int getPlayerId() {
        return playerId;
    }

    public void setPlayerId(int playerId) {
        this.playerId = playerId;
    }

    public int getEventId() {
        return eventId;
    }

    public void setEventId(int eventId) {
        this.eventId = eventId;
    }

    public int getChatId() {
        return chatId;
    }

    public void setChatId(int chatId) {
        this.chatId = chatId;
    }

    public Date getJoinDate() {
        return joinDate;
    }

    public void setJoinDate(Date joinDate) {
        this.joinDate = joinDate;
    }
}

File: server/database/entities/Participation.php
<?php

class Participation {
    private $joinDate;
    private $playerId;
    private $eventId;
    private $chatId;
    
    public function getJoinDate() {
        return $this->joinDate;
    }

    public function getPlayerId() {
        return $this->playerId;
    }

    public function getEventId() {
        return $this->eventId;
    }

    public function getChatId() {
        return $this->chatId;
    }

    public function setJoinDate($joinDate): void {
        $this->joinDate = $joinDate;
    }

    public function setPlayerId($playerId): void {
        $this->playerId = $playerId;
    }

    public function setEventId($eventId): void {
        $this->eventId = $eventId;
    }

    public function setChatId($chatId): void {
        $this->chatId = $chatId;
    }
    
    public static function fromObject($obj) {
        $participation = new Participation();
        if (isset($obj->joinDate)) {
            $participation->joinDate = $obj->joinDate;
        }
        if (isset($obj->playerId)) {
            $participation->playerId = $obj->playerId;
        }
        if (isset($obj->eventId)) {
            $participation->eventId = $obj->eventId;
        }
        if (isset($obj->chatId)) {
            $participation->chatId = $obj->chatId;
        }
        return $participation;
    }

}